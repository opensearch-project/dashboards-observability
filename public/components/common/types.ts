/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export interface FieldCapAttributes {
  type: string;
  searchable: boolean;
  aggregatable: boolean;
}

export interface FieldCapResponse {
  indices: string[];
  fields: Record<string, Record<string, FieldCapAttributes>>;
}
