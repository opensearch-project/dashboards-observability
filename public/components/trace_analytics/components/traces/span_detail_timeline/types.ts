/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { Span } from '../../common/constants';

export interface SpanWithChildren extends Span {
  children: SpanWithChildren[];
}

export interface SpanReference {
  refType: 'CHILD_OF' | 'FOLLOWS_FROM';
  spanID: string;
}

export type JaegerSpan = Omit<Span, 'spanId'> & {
  spanID: string;
  duration: number;
  references: SpanReference[];
};

export type SpanMap = Record<string, SpanWithChildren>;
