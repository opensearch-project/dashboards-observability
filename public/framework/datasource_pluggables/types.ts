/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { IDataFetcher } from '../../services/data_fetchers/fetch_interface';

export interface IDataSourceComponentSet {
  ui: {
    QueryEditor: React.ReactNode;
    ConfigEditor: React.ReactNode;
    SidePanel: React.ReactNode;
  };
  services: {
    data_fetcher: IDataFetcher;
  };
}

export interface IDataSourcePluggableComponents {
  languages?: Record<string, IDataSourceComponentSet>;
  // Other variation keys can be added in the future
}
