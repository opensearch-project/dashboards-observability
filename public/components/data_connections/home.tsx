/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { HashRouter, Route, RouteComponentProps, Switch } from 'react-router-dom';
import { ChromeBreadcrumb, ChromeStart, HttpStart } from '../../../../../src/core/public';
import { DataConnection } from './components/data_connection';
import { ManageDataConnectionsTable } from './components/manage_data_connections_table';

export interface HomeProps extends RouteComponentProps {
  pplService: any;
  parentBreadcrumb: ChromeBreadcrumb;
  http: HttpStart;
  chrome: ChromeStart;
}

export const Home = (props: HomeProps) => {
  const { http, chrome, pplService } = props;

  const commonProps = {
    http,
    chrome,
    pplService,
  };

  return (
    <HashRouter>
      <Switch>
        <Route
          exact
          path={'/manage/:id+'}
          render={(routerProps) => (
            <DataConnection
              {...commonProps}
              dataSource={decodeURIComponent(routerProps.match.params.id)}
            />
          )}
        />

        <Route
          exact
          path={['/', '/manage']}
          render={(routerProps) => <ManageDataConnectionsTable {...commonProps} />}
        />
      </Switch>
    </HashRouter>
  );
};
