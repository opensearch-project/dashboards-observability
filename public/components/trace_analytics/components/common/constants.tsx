/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

// Conversion factor for nanoseconds to milliseconds
export const NANOS_TO_MS = 1e6;

export const MILI_TO_SEC = 1000;

export const pieChartColors = [
  '#7492e7',
  '#c33d69',
  '#2ea597',
  '#8456ce',
  '#e07941',
  '#3759ce',
  '#ce567c',
  '#9469d6',
  '#4066df',
  '#da7596',
];

export interface Span {
  traceId: string;
  spanId: string;
  traceState: string;
  parentSpanId: string;
  name: string;
  kind: string;
  startTime: string;
  endTime: string;
  durationInNanos: number;
  serviceName: string;
  events: any[];
  links: any[];
  droppedAttributesCount: number;
  droppedEventsCount: number;
  droppedLinksCount: number;
  traceGroup: string;
  traceGroupFields: {
    endTime: string;
    durationInNanos: number;
    statusCode: number;
  };
  status: {
    code: number;
  };
  instrumentationLibrary: {
    name: string;
    version: string;
  };
}

export interface ParsedHit {
  _index: string;
  _id: string;
  _score: number;
  _source: Span;
  sort?: any[];
}
