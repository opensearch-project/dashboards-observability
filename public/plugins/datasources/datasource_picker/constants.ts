/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export const SOURCE_PICKER_TITLE = 'available sources';
export const SOURCE_PICKER_BTN_DEFAULT_TEXT = 'Select a datasource';
export const SOURCE_PICKER_BTN_DEFAULT_WIDTH = '300px';
export const SOURCE_PICKER_PANEL_DEFAULT_WIDTH = '500px';
export const SOURCE_PICKER_PANEL_SEARCH_TEST_SUBJ = 'selectableDatasourcePanelSearch';
export const SOURCE_PICKER_PANEL_TEST_SUBJ = 'selectableDatasourcePanel';
export const SOURCE_PICKER_FOOTER_CANCEL_BTN_TEXT = 'Cancel';
export const SOURCE_PICKER_FOOTER_SELECT_BTN_TEXT = 'Select';
export const SOURCE_PICKER_FOOTER_SELECT_BTN_TEST_SUBJ = 'datasourcePickerSelect';
export const SOURCE_PICKER_BTN_TEST_SUBJ = 'sourcePickerButton';
