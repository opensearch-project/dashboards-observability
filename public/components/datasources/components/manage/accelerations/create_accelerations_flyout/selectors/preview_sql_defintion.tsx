/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import {
  EuiAccordion,
  EuiButton,
  EuiCodeBlock,
  EuiFlexGroup,
  EuiFlexItem,
  EuiSpacer,
  EuiText,
} from '@elastic/eui';
import React, { useEffect, useState } from 'react';
import { CreateAccelerationForm } from '../../../../../../../../common/types/data_connections';
import { formValidator, hasError } from '../create/utils';
import { accelerationQueryBuilder } from '../visual_editors/query_builder';

interface PreviewSQLDefinitionProps {
  accelerationFormData: CreateAccelerationForm;
  setAccelerationFormData: React.Dispatch<React.SetStateAction<CreateAccelerationForm>>;
}

export const PreviewSQLDefinition = ({
  accelerationFormData,
  setAccelerationFormData,
}: PreviewSQLDefinitionProps) => {
  const [isPreviewStale, setIsPreviewStale] = useState(false);
  const [isPreviewTriggered, setIsPreviewTriggered] = useState(false);
  const [sqlCode, setSQLcode] = useState('');

  const onClickPreview = () => {
    const errors = formValidator(accelerationFormData);
    if (hasError(errors)) {
      setAccelerationFormData({ ...accelerationFormData, formErrors: errors });
      return;
    }
    setSQLcode(accelerationQueryBuilder(accelerationFormData));
    setIsPreviewStale(false);
    setIsPreviewTriggered(true);
  };

  useEffect(() => {
    setIsPreviewStale(true);
  }, [accelerationFormData]);

  return (
    <>
      <EuiAccordion
        id="accordion1"
        buttonContent={
          <EuiText data-test-subj="preview-sql-header">
            <h3>Preview SQL definition</h3>
          </EuiText>
        }
        paddingSize="l"
      >
        <EuiFlexGroup>
          <EuiFlexItem grow={false}>
            {isPreviewStale && isPreviewTriggered ? (
              <EuiButton
                iconType="kqlFunction"
                iconSide="left"
                color="success"
                onClick={onClickPreview}
              >
                Update preview
              </EuiButton>
            ) : (
              <EuiButton color="success" onClick={onClickPreview}>
                Generate preview
              </EuiButton>
            )}
          </EuiFlexItem>
          <EuiFlexItem grow={false}>
            <EuiButton iconType="popout" iconSide="right">
              Open in Query Workbench
            </EuiButton>
          </EuiFlexItem>
        </EuiFlexGroup>
        <EuiSpacer size="l" />
        <EuiCodeBlock language="sql" fontSize="m" paddingSize="m" isCopyable>
          {sqlCode}
        </EuiCodeBlock>
      </EuiAccordion>
    </>
  );
};
