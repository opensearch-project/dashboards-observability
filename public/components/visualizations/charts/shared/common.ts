/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { isEmpty, forEach, mapKeys } from 'lodash';
import { CUSTOM_LABEL } from '../../../../../common/constants/explorer';
import {
  ConfigList,
  DimensionSpan,
  VisSpecificMetaData,
} from '../../../../../common/types/explorer';
import { removeBacktick } from '../../../../../common/utils';

interface IIntermediateMapping {
  value: number;
  x: string;
  breakdown: string;
  aggName: string;
}

type PlotlyTrace = Partial<Plotly.Data>;

export const getCompleteTimespanKey = (span: DimensionSpan) => {
  if (isEmpty(span) || isEmpty(span.time_field) || isEmpty(span.interval) || isEmpty(span.unit))
    return '';
  const spanName = `span(${span.time_field[0]?.name},${span.interval}${span.unit[0]?.value})`;
  return { name: spanName, label: spanName };
};

/**
 * Transform to traces that can be consumed by plotly.
 * @param intermediateVisData intermediate visualization mapping data.
 * @param configList visualization configurations from config panel UI.
 * @param visTypeMetaData contains info related to how to prepare traces for this specifc visualizaton type.
 * @returns plotly traces.
 */
export const transformPreprocessedDataToTraces = (
  intermediateVisData: IIntermediateMapping[],
  { breakdowns, isVertical = true }: Partial<ConfigList>,
  visTypeMetaData: VisSpecificMetaData
): PlotlyTrace[] => {
  const traceMap = new Map<string, any>();
  const hasBreakdown = !isEmpty(breakdowns);
  forEach(intermediateVisData, (entry) => {
    const traceKey = hasBreakdown ? [entry.breakdown, entry.aggName].join(',') : entry.aggName;
    const xCoordinate = visTypeMetaData.x_coordinate;
    const yCoordinate = visTypeMetaData.y_coordinate;

    if (isEmpty(traceMap.get(traceKey))) {
      traceMap.set(traceKey, {
        [xCoordinate]: isVertical ? [entry.x] : [entry.value],
        [yCoordinate]: isVertical ? [entry.value] : [entry.x],
        name: hasBreakdown ? [entry.breakdown, entry.aggName].join(',') : `${traceKey}`,
      });
    } else {
      const curTrace = traceMap.get(traceKey);
      const xaxisValue = isVertical ? entry.x : entry.value;
      const yaxisValue = isVertical ? entry.value : entry.x;
      curTrace![xCoordinate].push(xaxisValue);
      curTrace![yCoordinate].push(yaxisValue);
    }
  });
  return [...traceMap.values()];
};

export const removeBackTick = (entry: any) => {
  return {
    ...mapKeys(entry, (val: any, key: string) => removeBacktick(key)),
  };
};

/**
 * preprocess raw schema-data, key-value mapping to form an intermediate,
 * dimension - breakdown - aggregation, key-value mapping.
 * 1. concatenate dimensions to generate one dimension
 * 2. concatenate breakdowns (if there's any) to generate one breakdown
 * 3. map dimension/breakdown to aggregations
 * @param jdbcFieldValueMapList raw jsonData comes from fetched data that has schema to data mapping
 * @param configList visualization configurations from config panel UI.
 * @returns intermediate visualization mapping data
 */
export const preprocessJsonData = (
  jdbcFieldValueMapList: any[],
  { dimensions, series, breakdowns, span }: Partial<ConfigList>
): IIntermediateMapping[] => {
  const seriesFlattenedEntries: IIntermediateMapping[] = [];
  forEach(jdbcFieldValueMapList, (entry: any) => {
    const backtickRemovedEntry = {
      ...removeBackTick(entry),
    };

    forEach(series, (sr) => {
      let tabularVizData: IIntermediateMapping = {
        value: 0,
        x: '',
        breakdown: '',
        aggName: '',
      };
      const serieKey = sr[CUSTOM_LABEL] ? sr[CUSTOM_LABEL] : `${sr.aggregation}(${sr.name})`;
      if (!isEmpty(serieKey)) {
        const concatedXaxisLabel = [
          ...(!isEmpty(span) ? [getCompleteTimespanKey(span)] : []),
          ...dimensions,
        ]
          .map((dimension) => {
            return backtickRemovedEntry[removeBacktick(dimension.name)] ?? '';
          })
          .join(',');
        const concatedBreakdownLabel = breakdowns
          ? breakdowns
              .map((breakdown) => backtickRemovedEntry[removeBacktick(breakdown.name)])
              .join(',')
          : '';
        tabularVizData = {
          value: backtickRemovedEntry[serieKey],
          x: concatedXaxisLabel,
          breakdown: concatedBreakdownLabel,
          aggName: serieKey,
        };
      }
      seriesFlattenedEntries.push(tabularVizData);
    });
  });
  return seriesFlattenedEntries;
};
