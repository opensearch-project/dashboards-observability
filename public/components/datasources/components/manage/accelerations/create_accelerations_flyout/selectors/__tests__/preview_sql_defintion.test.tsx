/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { waitFor } from '@testing-library/dom';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import React from 'react';
import { queryWorkbenchPluginCheck } from '../../../../../../../../../common/constants/shared';
import { CreateAccelerationForm } from '../../../../../../../../../common/types/data_connections';
import {
  coveringIndexBuilderMock1,
  createAccelerationEmptyDataMock,
  materializedViewBuilderMock2,
} from '../../../../../../../../../test/accelerations';
import { PreviewSQLDefinition } from '../preview_sql_defintion';

// @ts-ignore
global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () =>
      Promise.resolve({
        status: {
          statuses: [{ id: queryWorkbenchPluginCheck }],
        },
      }),
  })
);

describe('Preview SQL acceleration components', () => {
  configure({ adapter: new Adapter() });

  it('renders Preview SQL settings with default options', async () => {
    const accelerationFormData = createAccelerationEmptyDataMock;
    const setAccelerationFormData = jest.fn();
    const wrapper = mount(
      <PreviewSQLDefinition
        accelerationFormData={accelerationFormData}
        setAccelerationFormData={setAccelerationFormData}
      />
    );
    wrapper.update();
    await waitFor(() => {
      expect(
        toJson(wrapper, {
          noKey: false,
          mode: 'deep',
        })
      ).toMatchSnapshot();
    });
  });

  it('renders Preview SQL settings with default covering index options', async () => {
    const accelerationFormData: CreateAccelerationForm = {
      ...coveringIndexBuilderMock1,
      accelerationIndexType: 'covering',
    };
    const setAccelerationFormData = jest.fn();
    const wrapper = mount(
      <PreviewSQLDefinition
        accelerationFormData={accelerationFormData}
        setAccelerationFormData={setAccelerationFormData}
      />
    );
    wrapper.update();
    await waitFor(() => {
      expect(
        toJson(wrapper, {
          noKey: false,
          mode: 'deep',
        })
      ).toMatchSnapshot();
    });
  });

  it('renders Preview SQL settings with materialized view options', async () => {
    const accelerationFormData: CreateAccelerationForm = {
      ...materializedViewBuilderMock2,
      accelerationIndexType: 'covering',
    };
    const setAccelerationFormData = jest.fn();
    const wrapper = mount(
      <PreviewSQLDefinition
        accelerationFormData={accelerationFormData}
        setAccelerationFormData={setAccelerationFormData}
      />
    );
    wrapper.update();
    await waitFor(() => {
      expect(
        toJson(wrapper, {
          noKey: false,
          mode: 'deep',
        })
      ).toMatchSnapshot();
    });
  });

  it('renders Preview SQL settings with error state', async () => {
    const accelerationFormData: CreateAccelerationForm = {
      ...materializedViewBuilderMock2,
      accelerationIndexType: 'invlalid_state',
    };
    const setAccelerationFormData = jest.fn();
    const wrapper = mount(
      <PreviewSQLDefinition
        accelerationFormData={accelerationFormData}
        setAccelerationFormData={setAccelerationFormData}
      />
    );
    wrapper.update();
    await waitFor(() => {
      expect(
        toJson(wrapper, {
          noKey: false,
          mode: 'deep',
        })
      ).toMatchSnapshot();
    });
  });
});
