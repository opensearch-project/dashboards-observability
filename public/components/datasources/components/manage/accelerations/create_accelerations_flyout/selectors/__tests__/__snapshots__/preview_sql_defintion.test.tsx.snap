// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preview SQL acceleration components renders Preview SQL settings with default covering index options 1`] = `
<div
  className="euiAccordion"
>
  <div
    className="euiAccordion__triggerWrapper"
  >
    <button
      aria-controls="accordion1"
      aria-expanded={false}
      className="euiAccordion__button"
      id="random_html_id"
      onClick={[Function]}
      type="button"
    >
      <span
        className="euiAccordion__iconWrapper"
      >
        <svg
          aria-hidden={true}
          className="euiIcon euiIcon--medium euiAccordion__icon"
          focusable="false"
          height={16}
          role="img"
          style={null}
          viewBox="0 0 16 16"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m5.157 13.069 4.611-4.685a.546.546 0 0 0 0-.768L5.158 2.93a.552.552 0 0 1 0-.771.53.53 0 0 1 .759 0l4.61 4.684c.631.641.63 1.672 0 2.312l-4.61 4.684a.53.53 0 0 1-.76 0 .552.552 0 0 1 0-.771Z"
            fillRule="nonzero"
          />
        </svg>
      </span>
      <span
        className="euiIEFlexWrapFix"
      >
        <div
          className="euiText euiText--medium"
          data-test-subj="preview-sql-header"
        >
          <h3>
            Preview SQL definition
          </h3>
        </div>
      </span>
    </button>
  </div>
  <div
    aria-labelledby="random_html_id"
    className="euiAccordion__childWrapper"
    id="accordion1"
    role="region"
    tabIndex={-1}
  >
    <div>
      <div
        className="euiAccordion__padding--l"
      >
        <div
          className="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
        >
          <div
            className="euiFlexItem euiFlexItem--flexGrowZero"
          >
            <button
              className="euiButton euiButton--success euiButton--small"
              disabled={false}
              onClick={[Function]}
              style={
                Object {
                  "minWidth": undefined,
                }
              }
              type="button"
            >
              <span
                className="euiButtonContent euiButton__content"
              >
                <span
                  className="euiButton__text"
                >
                  Generate preview
                </span>
              </span>
            </button>
          </div>
          <div
            className="euiFlexItem euiFlexItem--flexGrowZero"
          />
        </div>
        <div
          className="euiSpacer euiSpacer--l"
        />
        <div
          className="euiCodeBlock euiCodeBlock--fontMedium euiCodeBlock--paddingMedium euiCodeBlock--hasControls prismjs language-sql"
          style={Object {}}
        >
          <pre
            className="euiCodeBlock__pre euiCodeBlock__pre--whiteSpacePreWrap"
            style={Object {}}
            tabIndex={-1}
          >
            <code
              className="euiCodeBlock__code sql"
            />
          </pre>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Preview SQL acceleration components renders Preview SQL settings with default options 1`] = `
<div
  className="euiAccordion"
>
  <div
    className="euiAccordion__triggerWrapper"
  >
    <button
      aria-controls="accordion1"
      aria-expanded={false}
      className="euiAccordion__button"
      id="random_html_id"
      onClick={[Function]}
      type="button"
    >
      <span
        className="euiAccordion__iconWrapper"
      >
        <svg
          aria-hidden={true}
          className="euiIcon euiIcon--medium euiIcon-isLoading euiAccordion__icon"
          focusable="false"
          height={16}
          role="img"
          style={null}
          viewBox="0 0 16 16"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5.277 10.088c.02.014.04.03.057.047.582.55 1.134.812 1.666.812.586 0 1.84-.293 3.713-.88L9 6.212V2H7v4.212l-1.723 3.876Zm-.438.987L3.539 14h8.922l-1.32-2.969C9.096 11.677 7.733 12 7 12c-.74 0-1.463-.315-2.161-.925ZM6 2H5V1h6v1h-1v4l3.375 7.594A1 1 0 0 1 12.461 15H3.54a1 1 0 0 1-.914-1.406L6 6V2Z"
          />
        </svg>
      </span>
      <span
        className="euiIEFlexWrapFix"
      >
        <div
          className="euiText euiText--medium"
          data-test-subj="preview-sql-header"
        >
          <h3>
            Preview SQL definition
          </h3>
        </div>
      </span>
    </button>
  </div>
  <div
    aria-labelledby="random_html_id"
    className="euiAccordion__childWrapper"
    id="accordion1"
    role="region"
    tabIndex={-1}
  >
    <div>
      <div
        className="euiAccordion__padding--l"
      >
        <div
          className="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
        >
          <div
            className="euiFlexItem euiFlexItem--flexGrowZero"
          >
            <button
              className="euiButton euiButton--success euiButton--small"
              disabled={false}
              onClick={[Function]}
              style={
                Object {
                  "minWidth": undefined,
                }
              }
              type="button"
            >
              <span
                className="euiButtonContent euiButton__content"
              >
                <span
                  className="euiButton__text"
                >
                  Generate preview
                </span>
              </span>
            </button>
          </div>
          <div
            className="euiFlexItem euiFlexItem--flexGrowZero"
          />
        </div>
        <div
          className="euiSpacer euiSpacer--l"
        />
        <div
          className="euiCodeBlock euiCodeBlock--fontMedium euiCodeBlock--paddingMedium euiCodeBlock--hasControls prismjs language-sql"
          style={Object {}}
        >
          <pre
            className="euiCodeBlock__pre euiCodeBlock__pre--whiteSpacePreWrap"
            style={Object {}}
            tabIndex={-1}
          >
            <code
              className="euiCodeBlock__code sql"
            />
          </pre>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Preview SQL acceleration components renders Preview SQL settings with error state 1`] = `
<div
  className="euiAccordion"
>
  <div
    className="euiAccordion__triggerWrapper"
  >
    <button
      aria-controls="accordion1"
      aria-expanded={false}
      className="euiAccordion__button"
      id="random_html_id"
      onClick={[Function]}
      type="button"
    >
      <span
        className="euiAccordion__iconWrapper"
      >
        <svg
          aria-hidden={true}
          className="euiIcon euiIcon--medium euiAccordion__icon"
          focusable="false"
          height={16}
          role="img"
          style={null}
          viewBox="0 0 16 16"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m5.157 13.069 4.611-4.685a.546.546 0 0 0 0-.768L5.158 2.93a.552.552 0 0 1 0-.771.53.53 0 0 1 .759 0l4.61 4.684c.631.641.63 1.672 0 2.312l-4.61 4.684a.53.53 0 0 1-.76 0 .552.552 0 0 1 0-.771Z"
            fillRule="nonzero"
          />
        </svg>
      </span>
      <span
        className="euiIEFlexWrapFix"
      >
        <div
          className="euiText euiText--medium"
          data-test-subj="preview-sql-header"
        >
          <h3>
            Preview SQL definition
          </h3>
        </div>
      </span>
    </button>
  </div>
  <div
    aria-labelledby="random_html_id"
    className="euiAccordion__childWrapper"
    id="accordion1"
    role="region"
    tabIndex={-1}
  >
    <div>
      <div
        className="euiAccordion__padding--l"
      >
        <div
          className="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
        >
          <div
            className="euiFlexItem euiFlexItem--flexGrowZero"
          >
            <button
              className="euiButton euiButton--success euiButton--small"
              disabled={false}
              onClick={[Function]}
              style={
                Object {
                  "minWidth": undefined,
                }
              }
              type="button"
            >
              <span
                className="euiButtonContent euiButton__content"
              >
                <span
                  className="euiButton__text"
                >
                  Generate preview
                </span>
              </span>
            </button>
          </div>
          <div
            className="euiFlexItem euiFlexItem--flexGrowZero"
          />
        </div>
        <div
          className="euiSpacer euiSpacer--l"
        />
        <div
          className="euiCodeBlock euiCodeBlock--fontMedium euiCodeBlock--paddingMedium euiCodeBlock--hasControls prismjs language-sql"
          style={Object {}}
        >
          <pre
            className="euiCodeBlock__pre euiCodeBlock__pre--whiteSpacePreWrap"
            style={Object {}}
            tabIndex={-1}
          >
            <code
              className="euiCodeBlock__code sql"
            />
          </pre>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Preview SQL acceleration components renders Preview SQL settings with materialized view options 1`] = `
<div
  className="euiAccordion"
>
  <div
    className="euiAccordion__triggerWrapper"
  >
    <button
      aria-controls="accordion1"
      aria-expanded={false}
      className="euiAccordion__button"
      id="random_html_id"
      onClick={[Function]}
      type="button"
    >
      <span
        className="euiAccordion__iconWrapper"
      >
        <svg
          aria-hidden={true}
          className="euiIcon euiIcon--medium euiAccordion__icon"
          focusable="false"
          height={16}
          role="img"
          style={null}
          viewBox="0 0 16 16"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m5.157 13.069 4.611-4.685a.546.546 0 0 0 0-.768L5.158 2.93a.552.552 0 0 1 0-.771.53.53 0 0 1 .759 0l4.61 4.684c.631.641.63 1.672 0 2.312l-4.61 4.684a.53.53 0 0 1-.76 0 .552.552 0 0 1 0-.771Z"
            fillRule="nonzero"
          />
        </svg>
      </span>
      <span
        className="euiIEFlexWrapFix"
      >
        <div
          className="euiText euiText--medium"
          data-test-subj="preview-sql-header"
        >
          <h3>
            Preview SQL definition
          </h3>
        </div>
      </span>
    </button>
  </div>
  <div
    aria-labelledby="random_html_id"
    className="euiAccordion__childWrapper"
    id="accordion1"
    role="region"
    tabIndex={-1}
  >
    <div>
      <div
        className="euiAccordion__padding--l"
      >
        <div
          className="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
        >
          <div
            className="euiFlexItem euiFlexItem--flexGrowZero"
          >
            <button
              className="euiButton euiButton--success euiButton--small"
              disabled={false}
              onClick={[Function]}
              style={
                Object {
                  "minWidth": undefined,
                }
              }
              type="button"
            >
              <span
                className="euiButtonContent euiButton__content"
              >
                <span
                  className="euiButton__text"
                >
                  Generate preview
                </span>
              </span>
            </button>
          </div>
          <div
            className="euiFlexItem euiFlexItem--flexGrowZero"
          />
        </div>
        <div
          className="euiSpacer euiSpacer--l"
        />
        <div
          className="euiCodeBlock euiCodeBlock--fontMedium euiCodeBlock--paddingMedium euiCodeBlock--hasControls prismjs language-sql"
          style={Object {}}
        >
          <pre
            className="euiCodeBlock__pre euiCodeBlock__pre--whiteSpacePreWrap"
            style={Object {}}
            tabIndex={-1}
          >
            <code
              className="euiCodeBlock__code sql"
            />
          </pre>
        </div>
      </div>
    </div>
  </div>
</div>
`;
